src/components/tx/ApprovalEditor/hooks/useApprovalInfos.ts

```diff
@@ -49,16 +55,22 @@ export const useApprovalInfos = (payload: {
             sameAddress(item.tokenInfo.address, approval.tokenAddress),
           )?.tokenInfo
 
+          let isErc721 = false
+
           if (!tokenInfo) {
-            tokenInfo = await getERC20TokenInfoOnChain(approval.tokenAddress)
+            try {
+              tokenInfo = await getERC20TokenInfoOnChain(approval.tokenAddress)
+            } catch (e) {
+              isErc721 = await isErc721Token(approval.tokenAddress)
```

Comment:
We should also try to fetch some minimal token information here. At least the name or symbol of the collection for instance.

Also we need to display the `id` that you are giving your approval to.
Currently the modal says "This allows the spender to transfer the specified token."
But we do not specify any token underneath.

Reply:
Good point, I added a call to fetch the symbol and decided to only fetch the symbol instead of the name or both because they are part of the same contract so either both exist or both don't and symbol is already part of the approval object.

I've also added a different message and included the token id. Let me know if this works for you:
<img width="659" alt="Screenshot 2024-07-26 at 11 38 12" src="https://github.com/user-attachments/assets/8d99b3d4-dd63-4451-8a94-8b8196022973">